base/applications/cleanmgr/dialog.c

```diff
@@ -0,0 +1,382 @@
+/*
+ * PROJECT:         ReactOS Disk Cleanup
+ * LICENSE:         GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:         Dialog Procs
+ * COPYRIGHT:       Copyright 2020 Arnav Bhatt (arnavbhatt288 at gmail dot com)
+ */
+ 
+#include "precomp.h"
+
+INT_PTR CALLBACK StartDlgProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    HBITMAP BitmapDrive = NULL;
+    switch(message)
+    {
+        case WM_INITDIALOG:
+        {
+            BitmapDrive = LoadBitmapW(GetModuleHandleW(NULL), MAKEINTRESOURCEW(IDB_DRIVE));
+ 
+            if (BitmapDrive == NULL)
+            {
+                MessageBoxW(NULL, L"LoadBitmapW() failed!", L"Error", MB_OK | MB_ICONERROR);
+                PostMessage(hwnd, WM_CLOSE, 0, 0);
+                return FALSE;
+            }
+            
+            InitStartDlg(hwnd, BitmapDrive);
+            return TRUE;
+        }
+
+        case WM_NOTIFY:
+            return ThemeHandler(hwnd, (LPNMCUSTOMDRAW)lParam);
+
+        case WM_THEMECHANGED:
+            InvalidateRect(hwnd, NULL, FALSE);
+            break;
+        
+        case WM_MEASUREITEM:
+        {
+            LPMEASUREITEMSTRUCT lpmis = (LPMEASUREITEMSTRUCT)lParam;
+
+            if (lpmis->itemHeight < 18 + 2)
+                lpmis->itemHeight = 18 + 2;
+
+            break;
+        }
+
+        case WM_DRAWITEM:
+        {
+            if(!DrawItemCombobox(lParam))
+            {
+                MessageBoxW(NULL, L"DrawItemCombobox() failed!", L"Error", MB_OK | MB_ICONERROR);
+                PostMessage(hwnd, WM_CLOSE, 0, 0);
+            }
+            break;
+        }
+
+        case WM_COMMAND:
+            if (HIWORD(wParam) == CBN_SELCHANGE)
+            {
+                int ItemIndex = SendMessageW(GetDlgItem(hwnd, IDC_DRIVE), CB_GETCURSEL, 0, 0);
+                WCHAR StringComboBox[ARR_MAX_SIZE] = { 0 };
+                WCHAR *GatheredDriveLatter = NULL;
+                if (ItemIndex == CB_ERR)
+                {
+                    MessageBoxW(NULL, L"SendMessageW failed!", L"Error", MB_OK | MB_ICONERROR);
+                    PostMessage(hwnd, WM_CLOSE, 0, 0);
+                }
+                SendMessageW(GetDlgItem(hwnd, IDC_DRIVE), CB_GETLBTEXT, (WPARAM)ItemIndex, (LPARAM)StringComboBox);
+                StringComboBox[wcslen(StringComboBox) - 1] = L'\0';
+                GatheredDriveLatter = wcsrchr(StringComboBox, L':');
+                GatheredDriveLatter--;
+                StringCbCopyW(wcv.DriveLetter, _countof(wcv.DriveLetter), GatheredDriveLatter);
```

Comment:
StringCbCopyW uses sizes in bytes, but you use _countof that returns size in "elements".